pkg/providers/amifamily/resolver.go

```diff
@@ -140,12 +140,14 @@ func (r Resolver) Resolve(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTem
 	return resolvedTemplates, nil
 }
 
-func GetAMIFamily(amiFamily *string, options *Options) AMIFamily {
+func GetAMIFamily(amiFamily *string, options *Options, template *v1alpha1.AWSNodeTemplate) AMIFamily {
 	switch aws.StringValue(amiFamily) {
 	case v1alpha1.AMIFamilyBottlerocket:
 		return &Bottlerocket{Options: options}
 	case v1alpha1.AMIFamilyUbuntu:
 		return &Ubuntu{Options: options}
+	case v1alpha1.AMIFamilyWindows:
+		return &Windows{Options: options, template: template}
```

Comment:
If we remove the annotation mechanism, then we remove the need for `template` to be passed through here

Reply:
Removed in the new code